Updates for Mojave and Catalina

smnogar opened the pull request:
I have had pretty good success getting this to build on recent versions of macOS. It has some specific versions of software like boost to increase the chances that this script continues to work.

Commit:
changes to rosdep

Commit:
Change to rosdeps

Commit:
rosdep

Commit:
rosdep

Commit:
Updated script to support gazebo8 and ogre1.9

smnogar committed:
Small changes to install script, rosdeps

Added a couple more dependencies to rosinstall, added boost@1.59 to rosdeps.yaml

smnogar committed:
default matplotlib back to pip

Commit:
updates for high sierra

Not much changed, update pips and do a brew upgrade and it should be good to go

Commit:
change gazebo8 install

Commit:
changed gazebo8 to default install

Commit:
changes to gazebo8 install

Commit:
Updated

Commit:
Merge branch 'high-sierra' of https://gitlab.sitcore.net/snogar/ros-install-macos into high-sierra

Commit:
More changes

Commit:
Moved to upstream geometry2

Commit:
changed rosdeps

Commit:
addex wxtools to rosdeps

Commit:
change to rosdep again

Commit:
more rosdep

Commit:
it finally worked

Commit:
Updated based on newest mikepurvis

Commit:
Merge remote-tracking branch 'origin-mikepurvis/master' into high-sierra

# Conflicts:
#	install
#	rosdeps.yaml

Commit:
updated readme

Commit:
more updates to readme

Commit:
Found fix for gazebo not working right

Commit:
Fixed link to correct version

Commit:
added custom rosdeps

Commit:
Added entry to ignore linux joystick drivers

Commit:
fixes for boost 1.67 and rosconsole

Commit:
fix custom build dir

Commit:
update for melodic as default

Commit:
fixes for kinetic and gazebo9 for melodic

Commit:
move some options from kinetic to melodic install

smnogar committed:
melodic updates

smnogar committed:
updates for mojave support

Commit:
changes to work in catalina

Still needs testing to see if works from scratch.

smnogar committed:
Update README.md

Commit:
fixes to improve chances it builds on first run

smnogar committed:
reduce chance that brew causes script failure

smnogar committed:
fallback to boost 1.69 to support mavros

smnogar committed:
Update README.md

Commit:
Use newest gazebo9

smnogar committed:
clean up comments

smnogar committed:
Update README.md

mikepurvis commented on README.md:
```diff
@@ -1,24 +1,59 @@
+* **opencv3**
+* **python2**
+* **sip**: 4.19.8_12 - Required for python2 support
+* **pyqt**: 5.10.1_1 - Required for python 2 support
```
How realistic would it be to have Python 3 as the default now? We're using it for most of the base system at CPR, so I'd expect it to mostly work.

smnogar replied:
I'm not sure. Isn't ROS planning on officially adopting it as of ROS "noetic" which isn't ready yet? I'll play around with it in a couple months before python 2 support officially gets dropped.

smnogar committed:
pcl and rospy fix. Use newest gazebo9

* Added a simple patch so that FindFLANN can find the flann library when building pcl_ros.
* Remove pcl_catkin
* Add patch for rospy to get rid of “concurrent poll() invocation" errors
* Build gazebo9 from scratch to help eliminate compatibility issues with protobuf

smnogar committed:
Merge branch 'catalina' of https://github.com/smnogar/ros-install-osx into catalina

smnogar committed:
fix patch command

ahoarau committed:
force link and verbose

if the link already exists it stops

smnogar committed:
Merge pull request #1 from ahoarau/patch-1

force link and verbose

smnogar committed:
add specific version of qt

smnogar committed:
a couple of patches for ros_controllers

Commit:
update to python3, add tango icons

upgrade to use python3, cleans out the manual installation of old dependencies that still had python2 support. Manually builds tango icons so that rqt has icons again.

smnogar committed:
fix typo

Commit:
fix rqt launch path

Commit:
add rqt launch path

Commit:
back to python2

splitting these branches up for the time being

smnogar committed:
Update README.md

smnogar committed:
more fixes…

smnogar committed:
latest changes

fjp commented:
Great work @smnogar! I hope this is merged one day and included in the documentation on how to install ros on osx.

smnogar committed:
small fix for numpy install and diagnostic_aggregator
